ts/features/bonus/cgn/saga/networking/activation/getBonusActivationSaga.ts

```diff
@@ -0,0 +1,97 @@
+import { cgnActivationStatus } from "../../../store/actions/activation";
+import { CgnActivationProgressEnum } from "../../../store/reducers/activation";
+import { startTimer } from "../../../../../../utils/timer";
+import { readablePrivacyReport } from "../../../../../../utils/reporters";
+import { CgnActivatedStatus } from "../../../../../../../definitions/cgn/CgnActivatedStatus";
+
+// wait time between requests
+const cgnResultPolling = 1000 as Millisecond;
+// stop polling when elapsed time from the beginning exceeds this threshold
+const pollingTimeThreshold = (10 * 1000) as Millisecond;
+
+const statusProgressRecord: Record<403 | 409, CgnActivationProgressEnum> = {
+  403: CgnActivationProgressEnum.INELIGIBLE,
+  409: CgnActivationProgressEnum.EXISTS
+};
+
+export type CgnStatusPollingSaga = ReturnType<typeof handleCgnStatusPolling>;
+
+export const cgnActivationSaga = (
+  startCgnActivation: ReturnType<typeof BackendCGN>["startCgnActivation"],
+  handleCgnStatusPolling: CgnStatusPollingSaga
+) =>
+  function* (): Generator<Effect, ActionType<typeof cgnActivationStatus>, any> {
+    try {
+      const startCgnActivationResult: SagaCallReturnType<typeof startCgnActivation> = yield call(
+        startCgnActivation,
+        {}
+      );
+
+      if (startCgnActivationResult.isRight()) {
+        const status = startCgnActivationResult.value.status;
+        // Status is 201 request has been created -> Start Polling
+        if (status === 201) {
+          return yield call(handleCgnStatusPolling);
+        }
+        // 202 -> still processing
+        if (status === 202) {
+          return cgnActivationStatus.success({
+            status: CgnActivationProgressEnum.PENDING
+          });
+        }
+        // 409 -> Cannot activate a new cgn because another card related to this user was found.
+        // 403 -> Ineligible
+        else if (status === 409 || status === 403) {
+          return cgnActivationStatus.success({
+            status: statusProgressRecord[status]
+          });
+        }
+        throw Error(`response status ${startCgnActivationResult.value.status}`);
+      }
+      // decoding failure
+      throw Error(readablePrivacyReport(startCgnActivationResult.value));
+    } catch (e) {
+      return cgnActivationStatus.failure(e);
+    }
+  };
+
+export const handleCgnStatusPolling = (
+  getCgnStatus: ReturnType<typeof BackendCGN>["getCgnStatus"]
+) =>
+  function* (): Generator<Effect, ActionType<typeof cgnActivationStatus>, any> {
+    const startPollingTime = new Date().getTime();
+    while (true) {
+      const cgnStatusResult: SagaCallReturnType<typeof getCgnStatus> = yield call(
+        getCgnStatus,
+        {}
+      );
+      // blocking error -> stop polling
+      if (cgnStatusResult.isLeft()) {
+        throw cgnStatusResult.value;
+      }
+      // we got the result -> stop polling
+      else if (
+        cgnStatusResult.isRight() &&
+        CgnActivatedStatus.is(cgnStatusResult.value.value)
```

Comment:
What about other possibile responses?
`CgnPendingStatus`, `CgnRevokedStatus`, `CgnExpiredStatus`
Should they be handled?

Now if one of these is returned by the API, it continues to polling and finishes with 
`status: CgnActivationProgressEnum.TIMEOUT`
But it could not be true

Reply:
Talking with the BE team, if card is in the activation process it can't be called any other procedure to revoke it.

So basically if we are in the activation process we will never receive a `CgnRevokedStatus` or `CgnExpiredStatus`

If it is instead `CgnPendingStatus` after the delay time the `status: CgnActivationProgressEnum.TIMEOUT` is proper to show the timeout screen, it means the user will be notified about the successfull activation with a message.

Reply:
Ok. 
Could it be a good idea to keep track of possibile unexpected responses?

```typescript
// we got the result -> stop polling
      else if (
        cgnStatusResult.isRight() &&
        cgnStatusResult.value.status === 200
      ) {
        if (CgnActivatedStatus.is(cgnStatusResult.value.value)) {
          return cgnActivationStatus.success({
            status: CgnActivationProgressEnum.SUCCESS,
            activation: cgnStatusResult.value.value
          });
        } else {
          void mixpanelTrack(getType(cgnActivationStatus.failure), {
            reason: `unexpected status result ${cgnStatusResult.value.value.status}`
          });
        }
      }
```